### ISSUE TITLE
`dialog` is not announced when the Popover opens (NVDA-specific)

### ISSUE THREAD

**Issue by Vinit-Pandit** [adobe/react-spectrum#9814]
### Provide a general summary of the issue here

When we open the Popover dialog, the word dialog is not announced in NVDA.

### 🤔 Expected Behavior?

The label along with the role dialog should be announced when the Popover dialog opens in NVDA.

### 😯 Current Behavior

The word dialog is not announced when the Popover dialog opens.

### 🖥️ Steps to Reproduce

The issue is reproducible on the Popover component: [site](https://react-aria.adobe.com/Popover)

https://github.com/user-attachments/assets/b51d3c4c-91b1-432a-b74d-941d7ae8ecee

### Version

Latest

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

Windows 11

**Comment by snowystinger** [comment]
I wonder if it's deduplicating the announcement or something. The popover is labeled by the button, "Settings".
VO is announcing the dialog in both Chrome and Safari.

Have we checked if there's an NVDA bug? Can we reproduce the issue without React Aria Components?
Could it be this? https://github.com/nvaccess/nvda/issues/8620

**Comment by Vinit-Pandit** [comment]
Thank you for pointing out the NVDA issue. Adding autoFocus to an interactive element allows the dialog to be read correctly. However, when focus moves to the container first, the issue does occur ( https://github.com/nvaccess/nvda/issues/8620 ).

I’ve also noticed a problem on macOS: when using `autoFocus`, the reading focus is not updated properly if focus moves directly to the interactive element. So it can be fixed if there is delay on `autoFocus` 

https://github.com/user-attachments/assets/6480e683-856b-4f8b-bb6a-ba977bc08d32

**Comment by snowystinger** [comment]
Thanks for the additional issue. In Chrome + VO, opening the dialog with Space or Enter, it moves focus to the text input and announces it. In Safari + VO, opening the dialog with Space or Enter does move focus, but does not announce it. Instead, you must use the VO keys to open the dialog. CTRL+OPT+Space. This is a pretty common theme with Safari and VO, usually using the VO keys has better results.

**Comment by Vinit-Pandit** [comment]
> Thanks for the additional issue. In Chrome + VO, opening the dialog with Space or Enter, it moves focus to the text input and announces it. In Safari + VO, opening the dialog with Space or Enter does move focus, but does not announce it. Instead, you must use the VO keys to open the dialog. CTRL+OPT+Space. This is a pretty common theme with Safari and VO, usually using the VO keys has better results.

Thank you for this.
but it still does not work with the Dialog Trigger → Popover → Dialog.
https://codesandbox.io/p/sandbox/nxd482?file=%2Fsrc%2FApp.js%3A13%2C23

When opening it using VoiceOver keys (Option + Shift + Space), the reading focus does not shift to the first interactive element( which have `autoFocus` ).

**Comment by snowystinger** [comment]
It's working for me, what version of MacOS (26.3.1) and Chrome (146.0.7680.154) are you on?

We'd be very reluctant to add a delay, they can cause more issues and bugs than they solve.

**Comment by Vinit-Pandit** [comment]
https://github.com/user-attachments/assets/adc40d0f-6d3f-488e-a201-119b5bbd44d8


MAC( 26.3.1 ) , Safari and voiceOver

**Comment by snowystinger** [comment]
https://github.com/user-attachments/assets/79e2441e-05af-49cd-830b-8a32d9d6668f

**Comment by yihuiliao** [comment]
https://github.com/user-attachments/assets/9af82242-5f41-4a77-8573-24b4d3567d52

this is what i'm seeing. macOS 15.2, safari 18.2

**Comment by Vinit-Pandit** [comment]
> Screen.Recording.2026-03-26.at.10.42.11.AM.mov 
> this is what i'm seeing. macOS 15.2, safari 18.2

Yes I think this issue in the safari 26:
https://discussions.apple.com/thread/256161078?sortBy=rank

**Comment by snowystinger** [comment]
Thanks, I've taken that discussion and opened a bug with webkit. I'm closing this and have added the webkit issue to our tracker wiki https://github.com/adobe/react-spectrum/wiki/Tracker-for-External-Browser-Bugs,-Library-Bugs,-and-Features

**Comment by snowystinger** [comment]
FYI, the webkit bug has been closed https://bugs.webkit.org/show_bug.cgi?id=311125
Please try the next safari tech preview and OS update after it's been released
